File: ded/soot/polyglot-1.3.5/src/polyglot/ast/ClassLit.java
package polyglot.ast;

/**
 * A <code>ClassLit</code> represents a class literal expression. 
 * A class literal expressions is an expression consisting of the 
 * name of a class, interface, array, or primitive type followed by a period (.) 
 * and the token class. 
 */
public interface ClassLit extends Lit {
    TypeNode typeNode();
}

File: ded/soot/polyglot-1.3.5/src/polyglot/ext/jl/tests/ClassLit.jl
class ClassLit {
    void m() {
      Class c = Object.class;
      c = int.class;
      c = boolean.class;
      c = long.class;
      c = void.class;
      c = boolean[].class;
      c = Class[].class;
      c = ClassLit.class;
      c = ClassLitInt.class;
    }
}

interface ClassLitInt {}
